### PR TITLE
Added test for setDefaultAutoScalerProfile

### LINKED ISSUE #3431: Add missing test coverage for azuremanagedcontrolplane_default.go
/kind cleanup

**Describe the solution you'd like**
Currently, `azuremanagedcontrolplane_default.go` is missing unit tests in the following areas:

- [setDefaultAutoScalerProfile()](https://app.codecov.io/github/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azuremanagedcontrolplane_default.go#L87): Add unit test to ensure that the default auto scaler profile is set if all fields are nil.

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by samyakjain10**
**What type of PR is this?** Added missing code coverage

**What this PR does / why we need it**: This PR impoves code coverage

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3431 

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by willie-yao — APPROVED** [review]
/lgtm

Thanks for your contribution and working on this issue!

File: api/v1beta1/azuremanagedcontrolplane_default_test.go [hunk 1]
```diff
@@ -53,3 +54,36 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
...
+			AutoScalerProfile: &AutoScalerProfile{},
+		},
+	}
+
+	// Test when all fields are nil
+	m.setDefaultAutoScalerProfile()
+
+	g := NewWithT(t)
+
+	// Verify that the default values are set
```
Comment (CecileRobertMichon) [inline api/v1beta1/azuremanagedcontrolplane_default_test.go:70]: should we also add a test case to verify that custom values don't get overwritten?

Comment (willie-yao) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:70]: Agreed! Good catch

Comment (samyakjain10) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:70]: Should i add that test checking for all values or for some values?

Comment (willie-yao) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:70]: I would shoot for all values. I think that we can test passing in an autoscaler with all values populated by something other than the default, then ensuring the output matches.

Comment (samyakjain10) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go]: I have made the changes. Please review.

File: api/v1beta1/azuremanagedcontrolplane_default_test.go [hunk 2]
```diff
@@ -53,3 +54,86 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
 		},
 	}
 }
+
+func TestSetDefaultAutoScalerProfile(t *testing.T) {
+	t.Run("AllFieldsAreNil", func(t *testing.T) {
```
Comment (willie-yao) [inline api/v1beta1/azuremanagedcontrolplane_default_test.go:59]: I don't think it's entirely necessary to use these `t.Run()` calls and instead just use assertions. You would have to rename the `AzureManagedControlPlane` with different names. 

See this test for examples: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1beta1/azuremachinepool_default_test.go#L50

File: api/v1beta1/azuremanagedcontrolplane_default_test.go [hunk 3]
```diff
@@ -53,3 +54,86 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
...
+					ScaleDownUtilizationThreshold: pointer.String("0.4"),
+					SkipNodesWithLocalStorage:     (*SkipNodesWithLocalStorage)(pointer.String(string(SkipNodesWithLocalStorageTrue))),
+					SkipNodesWithSystemPods:       (*SkipNodesWithSystemPods)(pointer.String(string(SkipNodesWithSystemPodsFalse))),
+				},
+			},
+		}
+
+		g := NewWithT(t)
+
+		// Verify that the values are set as expected
```
[the remark this answers is not in the document]
Comment (samyakjain10) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:119]: Yes, that is a very good suggestion.

**Comment by samyakjain10** [comment]
@willie-yao I have made the necessary changes. Please review

File: api/v1beta1/azuremanagedcontrolplane_default_test.go [hunk 4]
```diff
@@ -53,3 +54,90 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
...
+		},
+	}}
+
+	allFieldsAreNilTest.amcp.setDefaultAutoScalerProfile()
+
+	g.Expect(allFieldsAreNilTest.amcp.Spec.AutoScalerProfile).To(Equal(defaultAMP.Spec.AutoScalerProfile))
+
+	allFieldsAreNotNilTest := test{amcp: &AzureManagedControlPlane{
+		Spec: AzureManagedControlPlaneSpec{
+			AutoScalerProfile: &AutoScalerProfile{
```
Comment (willie-yao) [inline api/v1beta1/azuremanagedcontrolplane_default_test.go:101]: I think it would make the code a bit cleaner to set the AutoScalerProfile to a variable, then compare that to the outcome to ensure none of the fields have changed. That would save a few lines of code and not require a comparison for each individual field.

Comment (samyakjain10) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:101]: Done. Thanks for the suggestion.

Comment (willie-yao) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:101]: Since the `AutoScalerProfile` is equivalent to `expectedNotNil`, it should work out to set `AutoScalerProfile` to expectedNotNil, then ensuring it is still equal to `expectedNotNil` after calling `setDefaultAutoScalerProfile()`. That way, you don't have to repeat setting the long struct.

**Comment by willie-yao** [comment]
@samyakjain10 Apologies for the delay. I just left one more comment to clean up the code a bit. Otherwise this looks perfect!

File: api/v1beta1/azuremanagedcontrolplane_default_test.go (hunk 4 above)
Comment (samyakjain10) [reply api/v1beta1/azuremanagedcontrolplane_default_test.go:101]: I have made the changes

**Comment by samyakjain10** [comment]
@willie-yao No worries. I have made the changes.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve
